[BUG] Nil pointer exceptions can happen if the charts for the etcd backups copy task are invalid

**Describe the bug:**
During reconciliation of an `EtcdCopyBackupsTask` if there is an error when rendering the required charts, a nil pointer exception occurs. Additionally, to crashing the process, this nil pointer exception will hide the original cause for the error, which is the invalid charts. 
The nil pointer can happen both on lines https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L220 and https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L226

**Expected behavior:**
Nil pointer exception to not occur and the proper error to be returned.

**Stack trace:**
```
2022-03-11T16:54:04.786+0530	DEBUG	etcd-copy-backups-task-controller	Reconciling creation or update	{"task": "dev/etcd-test", "operation": "reconcile"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x24fcf2a]

goroutine 241 [running]:
github.com/gardener/etcd-druid/controllers.setStatusDetails(0x0, 0x1, 0x0, {0x2b43340, 0xc00071d940})
	/Users/i539698/go/src/druid_forked/etcd-druid/controllers/etcdcopybackupstask_controller.go:332 +0x4a
github.com/gardener/etcd-druid/controllers.(*EtcdCopyBackupsTaskReconciler).doReconcile.func1()
	/Users/i539698/go/src/druid_forked/etcd-druid/controllers/etcdcopybackupstask_controller.go:199 +0x3d
github.com/gardener/etcd-druid/controllers.(*EtcdCopyBackupsTaskReconciler).doReconcile(0xc00091eff0, {0x2b7fc28, 0xc000b6dad0}, 0xc000732fc0, {0x2b9f040, 0xc000630060})
	/Users/i539698/go/src/druid_forked/etcd-druid/controllers/etcdcopybackupstask_controller.go:220 +0x3d1
github.com/gardener/etcd-druid/controllers.(*EtcdCopyBackupsTaskReconciler).reconcile(0xc00091eff0, {0x2b7fc28, 0xc000b6dad0}, 0xc000732fc0)
	/Users/i539698/go/src/druid_forked/etcd-druid/controllers/etcdcopybackupstask_controller.go:147 +0x439
github.com/gardener/etcd-druid/controllers.(*EtcdCopyBackupsTaskReconciler).Reconcile(0xc00091eff0, {0x2b7fc28, 0xc000b6dad0}, {{{0xc000461369, 0x279bce0}, {0xc000461360, 0xc0007c2c00}}})
	/Users/i539698/go/src/druid_forked/etcd-druid/controllers/etcdcopybackupstask_controller.go:123 +0x165
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0004a6dc0, {0x2b7fc28, 0xc000b6da40}, {{{0xc000461369, 0x279bce0}, {0xc000461360, 0xc000649280}}})
	/Users/i539698/go/src/druid_forked/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114 +0x222
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004a6dc0, {0x2b7fb80, 0xc000649240}, {0x2714480, 0xc00071c080})
	/Users/i539698/go/src/druid_forked/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311 +0x2f2
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004a6dc0, {0x2b7fb80, 0xc000649240})
	/Users/i539698/go/src/druid_forked/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/Users/i539698/go/src/druid_forked/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/Users/i539698/go/src/druid_forked/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:223 +0x354
```

**Environment (please complete the following information):**
 - Etcd-druid version/commit ID : latest master branch

**Anything else we need to know?**:
The issue is caused because when [doReconcile](https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L194) method is completed the fields of the status var are updated in a `defer()` call, but some code paths directly return nil, instead of returning the initialised status variable. Check the following lines: [example1](https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L220), [example2](https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L226)
All code paths should be updated to return the already initialised status variable, similar to how it is done in [doDelete](https://github.com/gardener/etcd-druid/blob/fefe1cf9c3e00aa571020f9ddb8652af7691c5cc/controllers/etcdcopybackupstask_controller.go#L238)

/cc @ishan16696

--- a/controllers/etcdcopybackupstask_controller.go
+++ b/controllers/etcdcopybackupstask_controller.go
@@ -217,13 +217,13 @@ func (r *EtcdCopyBackupsTaskReconciler) doReconcile(ctx context.Context, task *d
 	// Render chart
 	renderedChart, err := r.chartApplier.Render(getEtcdCopyBackupsChartPath(), task.Name, task.Namespace, values)
 	if err != nil {
-		return nil, fmt.Errorf("could not render chart: %w", err)
+		return status, fmt.Errorf("could not render chart: %w", err)
 	}
 
 	// Decode job object from chart
 	job = &batchv1.Job{}
 	if err := decodeObject(renderedChart, getJobPath(), &job); err != nil {
-		return nil, fmt.Errorf("could not decode job object from chart: %w", err)
+		return status, fmt.Errorf("could not decode job object from chart: %w", err)
 	}
 
 	// Create job

